vnext/Microsoft.ReactNative/CoreApp.cpp

```diff
@@ -0,0 +1,306 @@
+            app->packageProvidersAbi[countAdded++] = packageProviderAbi;
+          } else {
+            assert(false && "Could not find native module factory function");
+          }
+        } else {
+          assert(false && "Could not find native module library");
+        }
+      }
+    }
+    app->packageProvidersAbiCount = countAdded;
+  }
+}
+
+static void SetProperties(const JsonObject &json, RNCoreApp *app) {
+  react::IReactPropertyBag bag{nullptr};
+  if (auto v = json.TryLookup(L"properties"); v && v.ValueType() == JsonValueType::Object) {
+    auto props = v.GetObject();
+    bag = react::ReactPropertyBagHelper::CreatePropertyBag();
+    for (const auto &prop : props) {
+      auto name = prop.Key();
+      auto lastDot = std::find(name.rbegin(), name.rend(), L'.');
+      std::wstring nsName;
+      std::wstring localName;
+      if (lastDot != name.rend()) {
+        nsName = std::wstring{name.begin(), static_cast<uint64_t>(name.rend() - lastDot - 1)};
+        localName = std::wstring{lastDot.base()};
+      } else {
+        localName = name;
+      }
+      winrt::hstring nsNameH(nsName);
+      winrt::hstring localNameH(localName);
+      auto propName = react::ReactPropertyName{nsNameH, localNameH};
+      auto value = prop.Value();
+      switch (value.ValueType()) {
+        case JsonValueType::String:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetString()));
+          break;
+        case JsonValueType::Boolean:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetBoolean()));
+          break;
+        case JsonValueType::Number:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetNumber()));
+          break;
+        case JsonValueType::Null:
+          bag.Set(propName.Handle(), nullptr);
+          break;
+        default:
+          assert(false && "not yet implemented");
+          break;
+      }
+    }
+  }
+
+  if (bag) {
+    winrt::copy_to_abi(bag, app->propertiesAbi);
+  }
+}
+
+extern "C" NORETURN void __cdecl RNStartCoreAppFromConfigJson(wchar_t const *configJson, coreAppCallback launched) {
+  auto app = new RNCoreApp();
+  RNCoreApp_SetDefaults(app);
+
+  auto json = LoadJsonFromFile(configJson);
+  SET_FROM_JSON(jsBundleFile);
+  SET_FROM_JSON(bundleRootPath);
+  SET_FROM_JSON(componentName);
+  SET_FROM_JSON(useWebDebugger);
+  SET_FROM_JSON(useFastRefresh);
+  SET_FROM_JSON(useDeveloperSupport);
+  SET_FROM_JSON(useDirectDebugger);
+  SET_FROM_JSON(requestInlineSourceMap);
+
+  SET_FROM_JSON(enableDefaultCrashHandler);
+  SET_FROM_JSON(debuggerPort);
+  SET_FROM_JSON(sourceBundlePort);
+  SET_FROM_JSON(sourceBundleHost);
+  SET_FROM_JSON(jsEngine);
+
+  SET_FROM_JSON(viewName);
+
+  SetPackageProviders(json, app);
+  SetProperties(json, app);
+  RNStartCoreAppWithCoreApp(app, launched);
+}
+
+NORETURN void __cdecl RNStartCoreAppWithModules(coreAppCallback launched, coreAppCallback addModules) {
+  auto rnca = std::make_shared<RNCoreApp>();
```

Comment:
>rnca [](http://example.com/codeflow?start=7&length=4)

nit: I wonder if we can use a better name - e.g. appSettings or appOptions. It is not easy to read and understand the new non-common abbreviation.